l2geth/core/genesis.go

```diff
@@ -387,59 +383,29 @@ func DefaultGoerliGenesisBlock() *Genesis {
-// Additional runtime parameters are passed through that impact
-// the genesis state. An "incompatible genesis block" error means that
-// these params were altered since the initial creation of the datadir.
-func DeveloperGenesisBlock(period uint64, faucet common.Address, stateDumpPath string, chainID *big.Int, gasLimit uint64) *Genesis {
+func DeveloperGenesisBlock(period uint64, faucet common.Address) *Genesis {
 	// Override the default period to the user requested one
 	config := *params.AllCliqueProtocolChanges
 	config.Clique.Period = period
 
-	if chainID != nil {
-		config.ChainID = chainID
-	}
-
-	stateDump := GenesisAlloc{}
-	if rcfg.UsingOVM {
-		// Fetch the state dump from the state dump path
-		// The system cannot start without a state dump as it depends on
-		// the ABIs that are included in the state dump. Check that all
-		// required state dump entries are present to prevent a faulty
-		// state dump from being used
-		if stateDumpPath == "" {
-			panic("Must pass state dump path")
-		}
-		log.Info("Fetching state dump", "path", stateDumpPath)
-		err := fetchStateDump(stateDumpPath, &stateDump)
-		if err != nil {
-			panic(fmt.Sprintf("Cannot fetch state dump: %s", err))
-		}
-	}
-
-	alloc := GenesisAlloc{
-		common.BytesToAddress([]byte{1}): {Balance: big.NewInt(1)}, // ECRecover
-		common.BytesToAddress([]byte{2}): {Balance: big.NewInt(1)}, // SHA256
-		common.BytesToAddress([]byte{3}): {Balance: big.NewInt(1)}, // RIPEMD
-		common.BytesToAddress([]byte{4}): {Balance: big.NewInt(1)}, // Identity
-		common.BytesToAddress([]byte{5}): {Balance: big.NewInt(1)}, // ModExp
-		common.BytesToAddress([]byte{6}): {Balance: big.NewInt(1)}, // ECAdd
-		common.BytesToAddress([]byte{7}): {Balance: big.NewInt(1)}, // ECScalarMul
-		common.BytesToAddress([]byte{8}): {Balance: big.NewInt(1)}, // ECPairing
-	}
-
-	for k, v := range stateDump {
-		alloc[k] = v
-	}
-
 	// Assemble and return the genesis with the precompiles and faucet pre-funded
 	return &Genesis{
 		Config:     &config,
 		ExtraData:  append(append(make([]byte, 32), faucet[:]...), make([]byte, crypto.SignatureLength)...),
-		GasLimit:   gasLimit,
+		GasLimit:   6283185,
 		Difficulty: big.NewInt(1),
-		Alloc:      alloc,
+		Alloc: map[common.Address]GenesisAccount{
+			common.BytesToAddress([]byte{1}): {Balance: big.NewInt(1)}, // ECRecover
+			common.BytesToAddress([]byte{2}): {Balance: big.NewInt(1)}, // SHA256
+			common.BytesToAddress([]byte{3}): {Balance: big.NewInt(1)}, // RIPEMD
+			common.BytesToAddress([]byte{4}): {Balance: big.NewInt(1)}, // Identity
+			common.BytesToAddress([]byte{5}): {Balance: big.NewInt(1)}, // ModExp
+			common.BytesToAddress([]byte{6}): {Balance: big.NewInt(1)}, // ECAdd
+			common.BytesToAddress([]byte{7}): {Balance: big.NewInt(1)}, // ECScalarMul
+			common.BytesToAddress([]byte{8}): {Balance: big.NewInt(1)}, // ECPairing
+			faucet:                           {Balance: new(big.Int).Sub(new(big.Int).Lsh(big.NewInt(1), 256), big.NewInt(9))},
```

Comment:
I don't think we want the faucet here, it'll introduce eth that wasn't deposited into the system

Reply:
We will no longer be running in developer mode, so having the faucet is fine